parser.py

```diff
@@ -1,17 +1,30 @@
 #!/bin/env python
 import argparse
 import pprint
+import re
+
 import grammar
 import visitor
 
+class PreProcessor:
+    @staticmethod
+    def remove_comments(program: str):
+        lines = program.splitlines()
+        cleaned_lines = []
+        for line in lines:
+            cleaned_line = re.sub(r'#.*$', '', line).strip()
```

Comment:
Don't strip lines. This will make the parser error reporting useless: you no longer know which column a problem happens at because whitespaces are removed from the original line.